### DISCUSSION TITLE
Combine common and tests

### DISCUSSION THREAD

**Discussion by inancgumus** [grafana/xk6-browser#143] [Ideas]
I propose combining the code and tests and restructuring the extension code.

**One of the reasons:**
The current package structure makes it hard to get coverage for the project.

Although we test the types in the `common/` package to some extent, the coverage tool cannot cover them. This is because the tests are in `tests/` while their tests are in `common/`. To do that, we need to use `-coverpkg` all the time.

This problem makes our jobs more complicated to create new tests and see what we are improving.

**Comment by mstoykov** [comment]
I would expect this is because the tests are in another package and by default coverage is only tracked for the package under test which is what `-coverpkg` is for. 

You can see how it's used in k6 [here](https://github.com/grafana/k6/blob/b8164eed5413732e94f0f334b9a873146bd0f58b/.github/workflows/all.yml#L146-L153) so that it also tracks coverage for all (k6) packages that are dependencies of the package under test.

**Comment by inancgumus** [comment]
Thanks, right, but I don't see a good reason for these super-related packages and their tests are in different packages.

```bash
go test ./tests ./common \
    -coverpkg=github.com/grafana/xk6-browser/common,github.com/grafana/xk6-browser/tests \
    -coverprofile=cover.out
```
